### PR TITLE
ceph: update helm version

### PR DESCRIPTION
**Pull request by binoue**
Currently, Helm v2.5.1 is used, but this version is released in 2017
and a little outdated.
To use the latest version of helm v2.x.x would be beneficial
as we can use more useful Helm commands like `helm template`.

This commit updates Helm version and stop using `helm lint` with `--strict` option 
because Helm v2.16.7 does not allow to use undefined values in values.yaml 
and the Helm upstream is now stopping using the `--strict` option.

Signed-off-by: binoue <banji-inoue@cybozu.co.jp>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **CommitLint Bot**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: cluster/charts/rook-ceph/templates/deployment.yaml [hunk 1]
```diff
@@ -116,109 +116,71 @@ spec:
...
           value: {{ .Values.csi.rbdPluginUpdateStrategy | quote }}
-{{- end }}
-{{- if .Values.csi.kubeletDirPath }}
         - name: ROOK_CSI_KUBELET_DIR_PATH
           value: {{ .Values.csi.kubeletDirPath | quote }}
-{{- end }}
         - name: ROOK_CSI_ENABLE_GRPC_METRICS
           value: {{ .Values.csi.enableGrpcMetrics | quote }}
-{{- if .Values.csi.cephcsi }}
-{{- if .Values.csi.cephcsi.image }}
```
Comment (travisn) [inline cluster/charts/rook-ceph/templates/deployment.yaml:130]: The expected approach for the CSI settings is that Rook has built-in defaults in the code. The settings should only be applied in the helm chart (or operator.yaml) if the non-defaults are desired by the admin. We don't want to set the defaults in the helm chart or else the upgrade process is not as smooth. Can you explain more why this change was made for the new helm?

File: cluster/charts/rook-ceph/values.yaml [hunk 2]
```diff
@@ -53,252 +53,268 @@ pspEnable: true
...
 ## needed
+csiEnable: true
 csi:
   enableRbdDriver: true
   enableCephfsDriver: true
   enableGrpcMetrics: true
   enableSnapshotter: true
   # (Optional) set user created priorityclassName for csi plugin pods.
-  # pluginPriorityClassName: system-node-critical
+  pluginPriorityClassName: system-node-critical
```
Comment (travisn) [inline cluster/charts/rook-ceph/values.yaml:63]: We don't know if these priority classes even exist so I don't think we can set a default, especially on k8s 1.14 or older before priority classes were available.

**Comment by binoue** [comment]
> > This commit updates Helm version and values.yaml
> > because helm v2.16.6 does not allow to use undefined values in values.yaml
> > when using helm lint --strict command.
> 
> Does this mean we just shouldn't use the `--strict` option? I don't think we can change our defaults and require them to all be specified.

I thought using `helm lint --strict` is better than `helm lint` because `--strict` option makes failing on lint warnings.
This behavior is not changed in the latest of helm v3.x.x.

So, I was trying to change the helm chart, but if the rook chart can not be changed to set all the default values explicitly, then using `lint` without `--strict` is the only choice .

**Comment by travisn** [comment]
> So, I was trying to change the helm chart, but if the rook chart can not be changed to set all the default values explicitly, then using lint without --strict is the only choice.

@binoue It would certainly be good to keep `--strict` if possible. Is there any way to suppress individual warnings? And can you post the output of the warnings here? Besides setting the default values, it would be good to confirm if there are any important to fix. Thanks!

**Comment by binoue** [comment]
@travisn 

Thank you for your advice!

I checked the docs. source codes, and issues in Helm.

>  Is there any way to suppress individual warnings?

In my understanding, there is no way to suppress individual warnings.

> And can you post the output of the warnings here?

Sure! The output below is the warnings.
```
$ helm version
Client: &version.Version{SemVer:"v2.16.7", GitCommit:"5f2584fd3d35552c4af26036f0c464191287986b", GitTreeState:"clean"}

$ helm lint ./cluster/charts/rook-ceph/
==> Linting ./cluster/charts/rook-ceph/
Lint OK

1 chart(s) linted, no failures

$ helm lint --strict ./cluster/charts/rook-ceph/
==> Linting ./cluster/charts/rook-ceph/
[ERROR] templates/: render error in "rook-ceph/templates/serviceaccount.yaml": template: rook-ceph/templates/_helpers.tpl:22:14: executing "imagePullSecrets" at <.Values.imagePullSecrets>: map has no entry for key "imagePullSecrets"

Error: 1 chart(s) linted, 1 chart(s) failed
```

However, through my research, I noticed that `--strict` option is only used for deciding the behavior for undefined values [here](https://github.com/helm/helm/blob/v2.16.7/pkg/engine/engine.go#L245-L251).
And this behavior is now removed in the master branch of Helm by [this PR](https://github.com/helm/helm/pull/8017).
(That is, `helm  lint --strict` is going to be the same as `helm lint`)

Considering the fact above, I think there is no positive reason for using `helm lint --strict`.
